src/main/kotlin/org/rust/cargo/runconfig/test/CargoTestLocator.kt

```diff
@@ -52,8 +50,8 @@ object CargoTestLocator : SMTestLocator {
             val name = qualifiedName.substringAfterLast(NAME_SEPARATOR)
             for (element in RsNamedElementIndex.findElementsByName(project, name, scope)) {
```

Comment:
Looks like it doesn't work correctly when lib and bin targets have items with the same names.
```rust
//- lib.rs
#[test]
fn foo() {}

//- main.rs
#[test]
fn foo() {}
```
In such case, we always navigate to single item (in my case from bin target).
Looks like we can't determine correct target in common case. And not sure that we want to do something special for items from doc comments (when we can filter out items from non-lib targets)

Reply:
I think this is a platform's issue, since `CargoTestLocator::getLocation` returns a list, but `SMTestProxy` always picks the first item from the list (see `SMTestProxy::computeLocation`):
```
List<Location> locations = locator.getLocation(protocolId, path, myMetainfo, project, searchScope);
return ContainerUtil.getFirstItem(locations);
```